docs/Contributing/architecture/system_design/database.md

```diff
@@ -0,0 +1,78 @@
+[Back to top](./README.md)
+# Database
+
+## MySQL
+
+Primary system of record. Schema evolved via Go migration files in `server/datastore/mysql/migrations/*` (tables + data). Access via datastore layer (`server/datastore/mysql/*.go`) implementing interface methods in `server/fleet` (e.g., `ListHosts`, `NewMDMAppleDeclaration`). SQL composed using `sqlx`, named queries, and small hand-written statements for performance-critical paths.
+
+### Setup / configuration
+Dev: docker-compose spins MySQL 8 with defaults. Test suites set `MYSQL_TEST=1` for integration tests; migrations auto-run on `New` datastore creation. Prod: configure pool sizes (`mysql.max_open_conns`, `max_idle_conns`, `conn_max_lifetime`), optional read replica (`replica` section) used for heavy read endpoints (software listings, vulnerability queries) – selected via `reader` context in datastore.
+
+### Connection lifetimes
+`config.MysqlConfig` holds DSN components; TLS optional (custom CA, certs). Long-running cron tasks keep short transactions; expensive bulk inserts (async host processing) use batches sized by config.
+
+### Host scale expectations
+| Hosts | Pattern | Notes |
+|-------|---------|-------|
+| <5k | synchronous host detail writes | OK to keep immediate updates |
+| 5k–50k | enable async host processing | Redis queue reduces write amplification |
+| 50k–200k | partition heavy tables logically via pruning queries (time windows) | Consider read replica |
+| 200k+ | archive / prune historical tables (software inventory history, activities) | Increase InnoDB buffer pool |
+
+### High-level entity groups
+* Inventory: `hosts`, `host_software`, `host_hardware`, `host_mdm_*`, `labels`, join tables for membership
+* Configuration: `app_config_json`, `teams`, `enroll_secrets`, `mdm_*_profiles`, `mdm_declarations`, `mdm_windows_configurations`
+* MDM State: `nano_enrollments`, `mdm_apple_commands`, `mdm_windows_commands`, results tables
+* Software / Vuln: `software`, `software_titles`, `software_cpe`, `cve_meta`, `software_vulnerabilities`
+* Scripts / Installers: `software_installers`, `scripts`, `script_executions`
+* Auth / Users: `users`, `api_tokens`, `sessions`, `invites`
+* Activities / Audit: `activities`, `activities_details`
+* Orchestration: `distributed_campaigns`, `distributed_query_executions` (live query tracking), `scheduled_queries`, `packs` (legacy)
```

Comment:
All names to be reviewed, I think `mdm_declarations` is wrong too, not sure about some of the others.